src/App.tsx

```diff
@@ -15,49 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = string;
+type Reverse = boolean;
+
+enum Sortfield {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+type Props = {
+  goods: Goods[];
+};
+
+const GoodsList: React.FC<Props> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
+function getPrepareGoods(
+  goods: Goods[],
+  sortGood: Sortfield | undefined,
+  reverse: Reverse,
+): Goods[] {
+  const prepareGoods: Goods[] = [...goods];
+
+  if (prepareGoods.length !== 0) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortGood) {
+        case Sortfield.Alphabet:
+          return good1.localeCompare(good2);
+
+        case Sortfield.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortGood, setSortGood] = useState<Sortfield>(Sortfield.Default);
+  const [reverse, setReverse] = useState<Reverse>(false);
+  const visibleGoods = getPrepareGoods(goodsFromServer, sortGood, reverse);
+
+  const getReset = (): void => {
```

Comment:
```suggestion
  const handleReset = (): void => {
```

Reply:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
Thank you this is really helpful